admin-api/resource_server/api-resources.md

```diff
@@ -0,0 +1,460 @@
+            scopes: ['api_client_write']
+        ),
+    ],
+    exceptionToStatus: [ApiClientNotFoundException::class => 404],
+)]
+class ApiClient
+{
+    #[ApiProperty(identifier: true)]
+    public int $apiClientId;
+}
+```
+
+## PaginatedList
+
+For listing operations we provided a custom operation based on the core grid system based on two settings:
+
+- **gridDataFactory**: service ID implementing the `GridDataFactoryInterface` (also used in all migrated pages listing)
+- **filtersClass**: the `Filters` class to used, it is optional but if you specify a custom one then you can force the default values
+
+{{% notice note %}}
+You can see this example also defines some custom mapping, you can read more about this bellow.
+{{% /notice %}}
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\APIResources\ApiPlatform\Resources\ApiClient;
+
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ApiResource;
+use PrestaShop\PrestaShop\Core\Search\Filters\ApiClientFilters;
+use PrestaShopBundle\ApiPlatform\Metadata\PaginatedList;
+
+#[ApiResource(
+    operations: [
+        new PaginatedList(
+            uriTemplate: '/api-clients',
+            scopes: [
+                'api_client_read',
+            ],
+            ApiResourceMapping: [
+                '[id_api_client]' => '[apiClientId]',
+                '[client_id]' => '[clientId]',
+                '[client_name]' => '[clientName]',
+                '[external_issuer]' => '[externalIssuer]',
+            ],
+            gridDataFactory: 'prestashop.core.grid.data_factory.api_client',
+            filtersClass: ApiClientFilters::class,
+            filtersMapping: [
+                '[apiClientId]' => '[id_api_client]',
+                '[clientId]' => '[client_id]',
+                '[clientName]' => '[client_name]',
+                '[externalIssuer]' => '[external_issuer]',
+            ],
+        ),
+    ],
+    normalizationContext: ['skip_null_values' => false],
+)]
+class ApiClientList
+{
+    #[ApiProperty(identifier: true)]
+    public int $apiClientId;
+
+    public string $clientId;
+
+    public string $clientName;
+
+    public string $description;
+
+    public ?string $externalIssuer;
+
+    public bool $enabled;
+
+    public int $lifetime;
+}
+```
+
+## Custom Mapping
+
+The API Platform DTO allows you to define the expected format of your API endpoint, including the way each field name format (snake case, camel case, ...) and any field you want to rename because it seems better.
+
+But the CQRS or Grid implementation may not match the format or naming you expected, which is why you can define some mapping to explain how to match the fields between your DTO and the underlying implementation.
+
+Each mapping is an associative array, the keys are the original naming and the value is the target mapping (the renamed key if you prefer), you can customize different mapping that are used in different moments of normalization in the workflow. 
```

Comment:
```suggestion
The API Platform DTO allows you to define the expected format of your API endpoint, including the format of each field name (snake case, camel case, etc.) and any field you want to rename because it seems better.

However, the CQRS or Grid implementation may not match the format or naming you expected, which is why you can define some mapping to explain how to match the fields between your DTO and the underlying implementation.

Each mapping is an associative array. The keys are the original naming, and the value is the target mapping (the renamed key if you prefer). You can customize different mappings that are used at different moments of normalization in the workflow. 
```